kotlinx-coroutines-core/common/src/flow/internal/ChannelFlow.kt

```diff
@@ -76,8 +85,11 @@ public abstract class ChannelFlow<T>(
      * handlers, while the pipeline before does not, because it was cancelled during its dispatch.
      * Thus `onCompletion` and `finally` blocks won't be executed and it may lead to a different kinds of memory leaks.
      */
-    open fun produceImpl(scope: CoroutineScope): ReceiveChannel<T> =
+    open fun produceImpl(scope: CoroutineScope): ReceiveChannel<T> = if (shouldDropWhileBusy) {
+        scope.dropWhileBusyProduce(context, start = CoroutineStart.ATOMIC, block = collectToFun)
```

Comment:
`produceCapacity` parameter is missing in the call to `dropWhileBusyProduce`. It should be affecting the chain of operators like  `dropWhileBusy().buffer(10)` where, logically, elements shall be dropped when the buffer is full.